packages/blade/src/components/Collapsible/CollapsiblePanel.web.tsx

```diff
@@ -0,0 +1,127 @@
+    transitionTimingFunction: castWebType(theme.motion.easing.standard.effective),
+    transitionProperty: 'height, opacity',
+    opacity: isExpanded ? 1 : 0.8,
+    /**
+     * We need height explicitly here for initial styles because the component might be rendered on server,
+     * in which case for expanded items this should be `auto` because we don't know the actual pixel value.
+     *
+     * This is based on `defaultIsExpanded` rather than `isExpanded` because post initial render inline styles take over.
+     * Otherwise, changing `height` both here and in inline styles sometimes causes animation flickers due to styles mismatch.
+     */
+    height: defaultIsExpanded ? HEIGHT_EXPANDED : HEIGHT_COLLAPSED,
+    display: defaultIsExpanded ? 'block' : 'none',
+    overflowY: 'hidden',
+  };
+});
+
+const CollapsiblePanel = ({ children }: CollapsiblePanelProps): ReactElement => {
+  const { isExpanded, defaultIsExpanded } = useCollapsibleContext();
+  const collapsiblePanelRef = useRef<HTMLDivElement>(null);
+
+  /**
+   * This effect imperatively updates height to make css transitions work:
+   * - for expanded items: auto height -> actual height -> 0px
+   * - for collapsed items: 0px -> actual height -> auto height
+   * - uses `requestAnimationFrame` to set the styles just before the next repaint
+   */
+  useDidUpdate(() => {
+    const collapsiblePanelElement = collapsiblePanelRef.current;
+
+    if (!collapsiblePanelElement) {
+      return;
+    }
+
+    // In collapsed state display is set to none, change it back to block
+    collapsiblePanelElement.style.display = 'block';
+    const actualHeight = collapsiblePanelElement.scrollHeight;
+
+    if (!isExpanded) {
+      // collapse
+      requestAnimationFrame(() => {
+        collapsiblePanelElement.style.height = makeSize(actualHeight);
+
+        requestAnimationFrame(() => {
+          collapsiblePanelElement.style.height = makeSize(0);
+        });
+      });
+    } else {
+      // expand
+      requestAnimationFrame(() => {
+        collapsiblePanelElement.style.height = makeSize(0);
+
+        requestAnimationFrame(() => {
+          collapsiblePanelElement.style.height = makeSize(actualHeight);
+
+          /**
+           * After this we want to wait for the animation to finish
+           * before setting the height back to auto
+           *
+           * `onTransitionEnd` takes over
+           */
+        });
+      });
+    }
+  }, [isExpanded]);
+
+  /**
+   * When expanding, waits for the animation to finish first.
+   * Then sets the height of expanded item to auto from actual height.
+   */
+  const onTransitionEnd: TransitionEventHandler = ({ propertyName }) => {
+    const collapsiblePanelElement = collapsiblePanelRef.current;
+    if (propertyName === 'height' && collapsiblePanelElement) {
+      if (isExpanded) {
+        // Panel has expanded and finished animating at this point
+        requestAnimationFrame(() => {
```

Comment:
what happens without RAF?

Reply:
Took a while to figure this one out 😅 

Without raf the animation would sometimes be jerky and I was also noticing occasional janks across browsers - Safari, Firefox, Brave. I think it has something to do with how React flushes updates, though I haven't digged enough yet to confirm.

Bottom line is these were needed to get smooth animations 100% of times across browsers.